client/src/app/shared/directives/only-number.directive.ts

```diff
@@ -0,0 +1,45 @@
+import { Directive, ElementRef, HostListener, Input } from '@angular/core';
+
+@Directive({
+    selector: '[osOnlyNumber]'
+})
+export class OnlyNumberDirective {
+    @Input()
+    public osOnlyNumber = true;
+
+    /**
+     * Regex to validate only numbers
+     * ^: starts with
+     * $: ends
+     */
+    private regExp = new RegExp('^[0-9]*$');
+
+    private allowedCharacters = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9'];
+
+    public constructor(private el: ElementRef) {}
+
+    @HostListener('keydown', ['$event'])
+    public onKeyDown(event: KeyboardEvent): void {
+        if (this.osOnlyNumber) {
+            if (
+                this.allowedCharacters.includes(event.key) ||
+                // Allow: Ctrl+A
+                (event.key === 'a' && event.ctrlKey) ||
```

Comment:
this may be too specific. what happens if i paste in characters? Maybe you have to additionally check *after* an event, if the modified text is ok. if not you can revert it to the last good known (which you can save in a member variable).